codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -189,8 +189,8 @@ class ServerHttpBoundProtocolTraitImplGenerator(
         "OnceCell" to RuntimeType.OnceCell,
         "PercentEncoding" to RuntimeType.PercentEncoding,
         "Regex" to RuntimeType.Regex,
-        "SmithyHttp" to RuntimeType.smithyHttp(runtimeConfig),
         "SmithyHttpServer" to ServerCargoDependency.smithyHttpServer(runtimeConfig).toType(),
+        "SmithyTypes" to RuntimeType.smithyTypes(runtimeConfig),
```

Comment:
nit: I am not 100% sure on how `.resolve` works. But I am assuming, that is the way a type should be referred to in the code generator instead of hard coding it like `#{SmithyTypes}::byte_stream::ByteStream`, which is how we've done in `streamingBodyTraitBounds`.

So instead of `SmithyTypes`, if we keep `"ByteStream" to RuntimeType.smithyTypes(rc).resolve("byte_stream::ByteStream")` in the codegenScope do you think that would resolve into the same generated code?

Reply:
yeah that results in the same generated code